homeassistant/components/shelly/switch.py

```diff
@@ -176,6 +182,14 @@ def async_setup_rpc_entry(
         RpcVirtualSwitch,
     )
 
+    async_setup_rpc_attribute_entities(
+        hass,
+        config_entry,
+        async_add_entities,
+        {"script": RPC_SCRIPT_SWITCH},
+        RpcScriptSwitch,
+    )
```

Comment:
If the user removes the script, the entity for it will remain in the registry. I suggest removing orphaned entities:

```python
    async_remove_orphaned_entities(
        hass,
        config_entry.entry_id,
        coordinator.mac,
        SWITCH_PLATFORM,
        coordinator.device.status,
        "script",
    )
```

Reply:
Good point, added